File: 26-removeDup.py
def removeDuplicates(arr): 
    pointer = 0 
    i = 0 
    while i < len(arr):
        if (i + 1 == len(arr) or arr[i] != arr[i + 1]):
            arr[pointer] = arr[i]
            pointer += 1
        i += 1
    return pointer

print(removeDuplicates([0,0,1,1,1,2,2,3,3,4]))

File: 26-removeDup.js
// var removeDuplicates = function(nums) {
//     for (let i = 0; i < nums.length - 1; i++) {
//         if (nums[i] === nums[i + 1]) {
//             nums.splice(i + 1, 1);
//             i--;
//         }
//     }
//     return nums;
// };
var removeDuplicates = function (nums) {
    let pointer = 0;
    for (let i = 0; i < nums.length; i++) {
        if (i + 1 === nums.length  || nums[i] !== nums[i + 1]) {
            nums[pointer] = nums[i];
            console.log(nums)
            pointer++;
        }
    }
    return pointer
}











console.log(removeDuplicates([0,0,1,1,1,2,2,3,3,4]))
